packages/client/tests/functional/reuse-binary-engine/tests.ts

```diff
@@ -0,0 +1,70 @@
+import * as util from 'util'
+
+import { setupTestSuiteMatrix } from '../_utils/setupTestSuiteMatrix'
+
+// https://github.com/prisma/prisma/issues/12507
+setupTestSuiteMatrix(() => {
+  test('should create data using one PrismaClient and read using another', async () => {
+    // @ts-ignore
+    const prismaClient1 = new PrismaClient({
```

Comment:
I think the type of `prismaClient1` will be `any` even in the generated tests if you do it like this (correct me if I'm wrong @millsp).

A better way would be to add

```ts
// @ts-ignore
declare let PrismaClient: typeof import('@prisma/client').PrismaClient
```

near the top of the file.

---

@millsp is there a way to avoid having to add these declarations manually in all tests? Maybe with a `.d.ts` file?

Reply:
Not sure how to type this one, I tried this however I get: 

```
Message:
      Argument of type '{ __internal: { engine: { endpoint: string; }; }; }' is not assignable to parameter of type 'Subset<PrismaClientOptions, PrismaClientOptions>'.
      Object literal may only specify known properties, and '__internal' does not exist in type 'Subset<PrismaClientOptions, PrismaClientOptions>'.
    at (tests/functional/reuse-binary-engine/.generated/reuse-binary-engine (sqlite) ()/tests.ts:39:7)

      47 |     for (const checkPath of keys(suiteChecks)) {
      48 |       if (checkPath.includes(path.dirname(suiteFilePath))) {
    > 49 |         assert.fail(suiteChecks[checkPath])
         |                ^
      50 |       }
      51 |     }
```

Reply:
@danstarns It does not exist on the types, it's on purpose. The best is to annotate with `// @ts-ignore`

Reply:
@aqrln I think that could be a good idea to explore along with making the types available while writing the test

Reply:
@danstarns `// @ts-ignore` should be applied on `__internal` and removed from the client instantiation and add
```ts
// @ts-ignore
declare let PrismaClient: typeof import('@prisma/client').PrismaClient
```